pkg/pillar/scripts/agent-start.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
+
+agent=$1
+shift
+
+# to start the agent inside service we touch the file
+while [ ! -f "/run/$agent.start" ]; do sleep 5; done
+
+# FIXME: this will need to be absorbed by containerd/shim
+keyctl link @u @s
+
+# Update our local /etc/hosts with entries coming from /config
+# We append on every boot since /etc/hosts starts from read-only rootfs
+[ -f /config/hosts ] && cat /config/hosts >> /etc/hosts
+
+uuid=$(cat $PERSISTDIR/status/uuid)
+/bin/hostname >/etc/hostname
+
```

Comment:
Can't we use bind mounts to avoid this in every service container?

Reply:
Yes, it is possible, modified